### PR TITLE
Elasticsearch: Only try once per client instance to get cluster name

### PR DESCRIPTION
**Pull request by tannalynn**
Instead of trying every time the client makes a request, lets only try once per client. If the agent fails to get a cluster name the first time, we will simply use nil and not try again, since it would probably fail again in the same way.

resolves https://github.com/newrelic/newrelic-ruby-agent/issues/2730

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # New Relic Ruby Agent Release Notes
 
 ## <dev>
-Version <dev> introduces instrumentation for the LogStasher gem and improves instrumentation for the `redis-clustering` gem.
+Version <dev> introduces instrumentation for the LogStasher gem, improves instrumentation for the `redis-clustering` gem, and updates the elasticsearch instrumentation to only attempt to get the cluster name once per client, even if it fails.
```
Comment (kaylareopelle) [inline CHANGELOG.md:4]:
```suggestion
Version <dev> introduces instrumentation for the LogStasher gem, improves instrumentation for the `redis-clustering` gem, and updates the Elasticsearch instrumentation to only attempt to get the cluster name once per client, even if it fails.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,11 @@ Version <dev> introduces instrumentation for the LogStasher gem and improves ins
...
 
+- **Bugfix: Update Elasticsearch instrumentation to only attempt to get the cluster name once per client**
+
+  Previously, the agent would attempt to get the cluster name every time a call was made if it was not already captured. This could lead to a large number of failures if the cluster name could not be retrieved. Now, the agent will only attempt to get the cluster name once per client, even if it fails. Thank you, [@ascoppa](https://github.com/ascoppa) for bringing this to our attention.
```
Comment (kaylareopelle) [inline CHANGELOG.md:16]: Could you link the issue/PR?

**Review by kaylareopelle — APPROVED** [review]
Two small changelog things, looks good otherwise!

File: CHANGELOG.md (hunk 2 above)
Comment (tannalynn) [reply CHANGELOG.md]: updated this! sorry, forgot about that part https://github.com/newrelic/newrelic-ruby-agent/pull/2743/commits/055f4438e9daf086c37d658ea9b7c76b10a55ed3

**Review by fallwith — APPROVED** [review]

**Review by hannahramadan — APPROVED** [review]

**Pull request merged**